.github/workflows/rust.yml

```diff
@@ -1,62 +1,119 @@
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
+      - 'test-ci/**'
+  pull_request:
 
 name: Continuous integration
 
 jobs:
-    tests:
-        name: Tests
-        runs-on: ubuntu-latest
-        strategy:
-            matrix:
-                include:
-                    - rust: stable
-                      env:
-                          RUSTFMTCHK: true
-                    - rust: nightly
-                      env:
-                          RUSTFMTCHK: false
-                    - rust: 1.56.1
-                      env:
-                          RUSTFMTCHK: false
-        steps:
-            - name: Checkout Crate
-              uses: actions/checkout@v2
-            - name: Checkout Toolchain
-              uses: actions-rs/toolchain@v1
-              with:
-                  profile: minimal
-                  toolchain: ${{ matrix.rust }}
-                  override: true
-            - name: Running test script
-              env: ${{ matrix.env }}
-              run: ./contrib/test.sh
+  Stable:                       # 2 jobs, one per lock file.
+    name: Test - stable toolchain
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        dep: [minimal, recent]
+    steps:
+      - name: "Checkout repo"
+        uses: actions/checkout@v4
+      - name: "Checkout maintainer tools"
+        uses: actions/checkout@v4
+        with:
+          repository: tcharding/rust-bitcoin-maintainer-tools
+          ref: 05-02-ci
```

Comment:
What happens once we loop an year?
This is not "collision resistant" 😂

Reply:
We also need to change the repository from tcharding/rbmt to rust-bicoin/rbmt. I was imagining then that the `ref` would just be a commit ID. But if we want to use tags then you're right, we should make sure the year is part of the name :)

Reply:
My bad, I did comment on that somewhere. Using my repo and branch is just for testing. 

> This is not "collision resistant" 😂

I'm currently solving this issue with a "clobber branches if they are a year old policy", its even more confusing when I accidentally use the wrong date a few days in the future